### PR TITLE
Add frame chunk method in labeling suggestions

### PR DESCRIPTION
**Pull request by hsingchien**
### Description
PR 3/3 for #977 
Added `frame chunk` method in labeling suggestions. User can add desired number of frame chunks for labeling suggestion. Then user can predict on these frames (select suggested frames in inference prompt). This is an enhancement to the issue discussed in #633 

### Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
[list open issues here]

### Outside contributors checklist

- [x] Review the [guidelines for contributing](https://github.com/talmolab/sleap/blob/develop/docs/CONTRIBUTING.md) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [x] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [x] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

**Comment by roomrys** [comment]
Hi @hsingchien,

Just checking in and seeing if this PR is ready for review?

Thanks,
Liezl

**Comment by hsingchien** [comment]
Marked ready for review. Sorry for the late response, I thought the PR draft would be pre-reviewed first before it is ready.

File: sleap/gui/suggestions.py [hunk 1]
```diff
@@ -275,6 +291,35 @@ def _velocity_video(
...
+        **kwargs,
+    ):
+        # check the validity of inputs, frame_from <= frame_to
+        if frame_from > frame_to:
+            return VideoFrameSuggestions.filter_unique_suggestions(labels, videos, [])
+
+        """Add consecutive frame chunk to label suggestion"""
+        proposed_suggestions = []
+        for video in videos:
+            # make sure when targeting all videos the from and to do not exceed frame number
```
Comment (roomrys) [inline sleap/gui/suggestions.py:310]: Nitpick: [google-style comments](https://google.github.io/styleguide/pyguide.html#386-punctuation-spelling-and-grammar)
```suggestion
            # Make sure when targeting all videos the from and to do not exceed frame number
```

File: sleap/gui/suggestions.py [hunk 2]
```diff
@@ -275,6 +291,35 @@ def _velocity_video(
...
+            if frame_from > video.num_frames:
+                continue
+            this_video_frame_to = min(frame_to, video.num_frames)
+            # generate list of frame numbers
```
Comment (roomrys) [inline sleap/gui/suggestions.py:314]: Nitpick: [google-style comments](https://google.github.io/styleguide/pyguide.html#386-punctuation-spelling-and-grammar)
```suggestion

            # Generate a list of frame numbers
```

**Review by roomrys — CHANGES_REQUESTED** [review]
Nice feature! Functionally, it works like a charm. Main comment is that I don't think we need a new `UpdateTopic` - we should just be able to use the `UpdateTopic.video`. There are a few more suggestions throughout. Let me know if you want any help implementing!

File: tests/gui/test_app.py [hunk 3]
```diff
@@ -74,22 +74,66 @@ def test_app_workflow(
 
     app.state["video"] = centered_pair_vid
 
+    # Verify the max of frame_chunk spinboxes is updated
+    method_layout = app.suggestions_form_widget.form_layout.fields["method"]
+    frame_chunk_layout = method_layout.page_layouts["frame chunk"]
+    frame_to_spinbox = frame_chunk_layout.fields["frame_to"]
+    frame_from_spinbox = frame_chunk_layout.fields["frame_from"]
+    cur_video = app.state["video"]
+    cur_video_framenum = cur_video.num_frames
```
[the remark this answers is not in the document]
Comment (hsingchien) [reply tests/gui/test_app.py]: Thanks for pointing out!

File: sleap/gui/app.py [hunk 4]
```diff
@@ -1321,6 +1330,17 @@ def _has_topic(topic_list):
...
+        if _has_topic([UpdateTopic.suggestions_ui]):
+            # update the max of <frame_chunk> spinbox to num_frames of current video
+            method_layout = self.suggestions_form_widget.form_layout.fields["method"]
+            frame_chunk_layout = method_layout.page_layouts["frame chunk"]
+            frame_to_spinbox = frame_chunk_layout.fields["frame_to"]
+            frame_from_spinbox = frame_chunk_layout.fields["frame_from"]
+            cur_video = self.state["video"]
+            cur_video_framenum = cur_video.num_frames
+            frame_to_spinbox.setMaximum(cur_video_framenum)
+            frame_from_spinbox.setMaximum(cur_video_framenum)
```
[the remark this answers is not in the document]
Comment (hsingchien) [reply sleap/gui/app.py]: Spinbox update is now under UpdateTopic.video

File: sleap/gui/suggestions.py [hunk 5]
```diff
@@ -275,6 +291,35 @@ def _velocity_video(
...
+        frame_from: int,
+        frame_to: int,
+        **kwargs,
+    ):
+        # check the validity of inputs, frame_from <= frame_to
+        if frame_from > frame_to:
+            return VideoFrameSuggestions.filter_unique_suggestions(labels, videos, [])
+
+        """Add consecutive frame chunk to label suggestion"""
+        proposed_suggestions = []
```
[the remark this answers is not in the document]
Comment (hsingchien) [reply sleap/gui/suggestions.py]: Doc string moved up. 
Validation failure now returns empty list.

File: tests/gui/test_suggestions.py [hunk 6]
```diff
@@ -166,6 +166,75 @@ def test_video_selection(
...
+    centered_pair_predictions_copy = centered_pair_predictions
+    # Clear existing suggestions so that generated suggestions will be kept intact at the uniqueness check step
+    centered_pair_predictions_copy.suggestions.clear()
+    suggestions = VideoFrameSuggestions.suggest(
+        labels=centered_pair_predictions_copy,
+        params={
+            "videos": centered_pair_predictions_copy.videos,
+            "method": "frame_chunk",
+            "frame_from": 1,
+            "frame_to": 1000,
```
[the remark this answers is not in the document]
Comment (hsingchien) [reply tests/gui/test_suggestions.py]: Changed to 20

**Pull request merged**